packages/rest/src/http/decorators.ts

```diff
@@ -563,17 +564,24 @@ export function getAuthentication(
  *
  * `@route` can only be applied to operations, namespaces, and interfaces.
  */
-export function $route(context: DecoratorContext, entity: Type, path: string) {
+export function $route(context: DecoratorContext, entity: Type, path: string, parameters?: Model) {
   setRoute(context, entity, {
     path,
     isReset: false,
+    shared: (parameters?.properties.get("shared")?.type as BooleanLiteral)?.value ?? false,
   });
 }
 
-export function $routeReset(context: DecoratorContext, entity: Type, path: string) {
+export function $routeReset(
+  context: DecoratorContext,
+  entity: Type,
+  path: string,
+  parameters?: Model
+) {
   setRoute(context, entity, {
     path,
     isReset: true,
+    shared: (parameters?.properties.get("shared")?.type as BooleanLiteral)?.value ?? false,
```

Comment:
https://github.com/microsoft/cadl/pull/1183#discussion_r999805958 still applies here.

That said, a quick search of uses of cadlTypeToJson suggests we have other bugs around this.

I commented on https://github.com/microsoft/cadl/pull/1161#issuecomment-1284463243 which would let the compiler defend this for us, but there's discussion to scale that back. 

I'm OK with tracking this general issue separately, but I don't think we should leave things in a state where you can pass the wrong types to our decorators.